ios/testmanagerd/testlistener.go

```diff
@@ -0,0 +1,74 @@
+	TestCaseDidStartForTestClassMethod()
+	TestCaseDidStartWithIdentifierTestCaseRunConfiguration()
+	TestMethodOfClassDidMeasureMetricFileLine()
+	TestSuiteDidFinishAtRunCountWithFailuresUnexpectedTestDurationTotalDuration()
+	TestSuiteWithIdentifierDidFinishAtRunCountSkipCountFailureCountExpectedFailureCountUncaughtExceptionCountTestDurationTotalDuration()
+	TestSuiteDidStartAt()
+	TestSuiteWithIdentifierDidStartAt(testIdentifier nskeyedarchiver.XCTTestIdentifier, date string)
+}
+
+type TestLogCollector struct {
+	Writer *io.Writer
+}
+
+func (t TestLogCollector) LogDebugMessage(msg string) {
+	(*t.Writer).Write([]byte(msg))
+}
+
+func (t TestLogCollector) LogMessage(msg string) {
+	(*t.Writer).Write([]byte(msg))
+}
+
+func (t TestLogCollector) DidBeginExecutingTestPlan()                                             {}
+func (t TestLogCollector) DidFinishExecutingTestPlan()                                            {}
+func (t TestLogCollector) DidFailToBootstrapWithError(err nskeyedarchiver.NSError)                {}
+func (t TestLogCollector) DidBeginInitializingForUITesting()                                      {}
+func (t TestLogCollector) GetProgressForLaunch()                                                  {}
+func (t TestLogCollector) InitializationForUITestingDidFailWithError(err nskeyedarchiver.NSError) {}
+func (t TestLogCollector) TestCaseMethodDidFinishActivity()                                       {}
+func (t TestLogCollector) TestCaseWithIdentifierDidFinishActivity(testIdentifier nskeyedarchiver.XCTTestIdentifier, activityRecord nskeyedarchiver.XCActivityRecord) {
+}
+func (t TestLogCollector) TestCaseMethodDidStallOnMainThreadInFileLine() {}
+func (t TestLogCollector) TestCaseMethodWillStartActivity()              {}
+func (t TestLogCollector) TestCaseWithIdentifierWillStartActivity(testIdentifier nskeyedarchiver.XCTTestIdentifier, activityRecord nskeyedarchiver.XCActivityRecord) {
+}
+func (t TestLogCollector) TestCaseDidFailForTestClassMethodWithMessageFileLine()   {}
+func (t TestLogCollector) TestCaseWithIdentifierDidRecordIssue()                   {}
+func (t TestLogCollector) TestCaseDidFinishForTestClassMethodWithStatusDuration()  {}
+func (t TestLogCollector) TestCaseWithIdentifierDidFinishWithStatusDuration()      {}
+func (t TestLogCollector) TestCaseDidStartForTestClassMethod()                     {}
+func (t TestLogCollector) TestCaseDidStartWithIdentifierTestCaseRunConfiguration() {}
+func (t TestLogCollector) TestMethodOfClassDidMeasureMetricFileLine()              {}
+func (t TestLogCollector) TestSuiteDidFinishAtRunCountWithFailuresUnexpectedTestDurationTotalDuration() {
+}
+func (t TestLogCollector) TestSuiteWithIdentifierDidFinishAtRunCountSkipCountFailureCountExpectedFailureCountUncaughtExceptionCountTestDurationTotalDuration() {
+}
+func (t TestLogCollector) TestSuiteDidStartAt() {}
+func (t TestLogCollector) TestSuiteWithIdentifierDidStartAt(testIdentifier nskeyedarchiver.XCTTestIdentifier, date string) {
```

Comment:
They're exported because `TestLogCollector` implements the interface `TestListener` and later an instance of it is passed as an argument in main.go to print test logs to stdout.

Reply:
Some of these callbacks contain really useful information about tests but many are indeed too verbose to make any use of. Do you think we can remove some of them entirely, both from `TestLogCollector` as well as the implemented `TestListener` interface?

Reply:
Also please see the up to date version of `TestListener`. Some callbacks now receive additional arguments as received from the device.

I'm open to remove any of those if they have no use.

Reply:
> is passed as an argument in main.go to print test logs to stdout while ignoring all other callbacks.

but what does `main.go` call on that interface? The functions are called by messages we get from the device and they report a status, but that won't happen in `main.go`. For example what would be a usecase for anything outside this package to call `DidBeginInitializingForUITesting`?

Reply:
`TestListener` is simpler now. Let me know how it looks.